cp3pt0-deployment/common/migrate_singleton.sh

```diff
@@ -81,6 +76,88 @@ function main() {
     success "Migration is completed for Cloud Pak 3.0 Foundational singleton services."
 }
 
+function migrate_license_service_reporter(){
+    title "LSR migration from ibm-cmmon-services to ${LSR_NAMESPACE}"
+
+    local lsr_instances=$("$OC" get IBMLicenseServiceReporter instance -n ibm-common-services --no-headers | wc -l)
+    if [[ lsr_instances -eq 0 ]]; then
+        info "No LSR for migration found in ibm-common-services namespace"
+        return 0
+    fi
+    # Prepare LSR PV/PVC which was decoupled in isolate.sh
+    # delete old LSR CR - PV will stay as during isolate.sh the policy was set to Retain
+    ${OC} delete IBMLicenseServiceReporter instance -n ibm-common-services
+
+    # in case PVC is blocked with deletion, the finalizer needs to be removed
+    lsr_pvcs=$("${OC}" get pvc license-service-reporter-pvc -n ibm-common-services  --no-headers | wc -l)
+    if [[ lsr_pvcs -gt 0 ]]; then
+        info "Failed to delete pvc license-service-reporter-pvc, patching its finalizer to null..."
+        ${OC} patch pvc license-service-reporter-pvc -n ibm-common-services  --type="json" -p '[{"op": "remove", "path":"/metadata/finalizers"}]'
+    else
+        debug1 "No pvc license-service-reporter-pvc as expected"
+    fi
+
+    lsr_pv_nr=$("${OC}" get pv -l license-service-reporter-pv=true --no-headers | wc -l )
+    if [[ lsr_pv_nr -gt 1 ]]; then
+        error "More than on PV with label license-service-reporter-pv=true was found. Only one is allowed."
+    fi
+
+    if [[ lsr_pv_nr -eq 1 ]]; then
+        debug1 "LSR namespace: ${LSR_NAMESPACE}" 
+        create_namespace "${LSR_NAMESPACE}"
+
+        # get storage class name
+        LSR_PV_NAME=$("${OC}" get pv -l license-service-reporter-pv=true -o=jsonpath='{.items[0].metadata.name}')
+        debug1 "PV name: $LSR_PV_NAME"
+        
+        # on ROKS storage class name cannot be proviced during PVC creation
+        roks=$(${OC} cluster-info | grep 'containers.cloud.ibm.com')
+        if [[ -z $roks ]]; then
+            LSR_STORAGE_CLASS=$("${OC}" get pv -l license-service-reporter-pv=true -o=jsonpath='{.items[0].spec.storageClassName}')
+            if [[ -z $LSR_STORAGE_CLASS ]]; then
+                error "Cannnot get storage class name from PVC license-service-reporter-pv in $LSR_NAMESPACE"
+            fi
+        else
+            debug1 "Run on ROKS, not setting storageclass name"
+            LSR_STORAGE_CLASS=""
+        fi
+
+        # create PVC
+        TEMP_LSR_PVC_FILE="_TEMP_LSR_PVC_FILE.yaml"
+
+        cat <<EOF >$TEMP_LSR_PVC_FILE
+        apiVersion: v1
+        kind: PersistentVolumeClaim
+        metadata:
+            name: license-service-reporter-pvc
+            namespace: ${LSR_NAMESPACE}
+        spec:
+            accessModes:
+            - ReadWriteOnce
+            resources:
+                requests:
+                    storage: 1Gi
+            storageClassName: "${LSR_STORAGE_CLASS}"
```

Comment:
Does empty storageClassName works on ROKS? Or it should not be added at all?

Reply:
We was about to check it. @martabereta could you comment on details?